src/main/java/com/sap/oss/phosphor/fosstars/tool/github/MarkdownReporter.java

```diff
@@ -130,14 +129,22 @@
 
     Objects.requireNonNull(rating, "Oh no! Rating is null");
     Objects.requireNonNull(advisor, "Oh no! Advisor is null");
-
     Objects.requireNonNull(outputDirectory, "Oh no! Output directory is null!");
-    if (!Files.isDirectory(Paths.get(outputDirectory))) {
-      throw new FileNotFoundException(
-          String.format("Oh no! I could not find %s", outputDirectory));
+
+    Path directory = Paths.get(outputDirectory);
+    if (Files.isSymbolicLink(directory) || Files.isRegularFile(directory)) {
+      throw new IOException(
+          String.format("Oops! Output directory is not a directory: %s", outputDirectory));
+    }
+    if (!Files.isDirectory(directory)) {
+      Files.createDirectories(directory);
+    }
+    if (!Files.isDirectory(directory)) {
```

Comment:
Isn't the condition on line https://github.com/SAP/fosstars-rating-core/pull/415/files#diff-0d1a21dc11fa7ecb0693a45578821f78d8b39558d2c0cdac1cdfaea7065aadf9R139 same as this?

Reply:
If the path is a symlink of a file, it doesn't make sense to continue. Then, the directory may or may not exist. Here we check it.

Reply:
No , I mean, just above this condition, we check the same and create right? then why again checking it?

Reply:
It makes sure that the directory was created. This check it a bit paranoid and therefore may look redundant.